### PR TITLE
Add `--analyze` param to CLI

### PR DESCRIPTION
**Pull request by matthewmturner**
TLDR: Execution plan output from EXPLAIN ANALYZE tells a story of how a query was executed and the goal of this command is to output a summary of that story. 

I am working on adding this feature because I often look at the output of EXPLAIN ANALYZE and need to connect several dots myself to see what is important which can be quite manual and then if you are trying to do this across different queries or with different source data layouts you end up redoing a lot of work. 

The new analyze parameter generates output like the below which is designed to make it easier to analyze queries and find hot spots (IO vs Compute and a breakdown of the core parts for each) and / or optimization opportunties.  It is still very much a WIP (organization, stats to shows, wording, etc) but it is now starting to take shape into something close to what i envision the v1 version will look like.

IO stats will have some relevant metrics available for different io execs (parquet, csv, ndjson, arrow, etc).  Starting with just parquet for now.

For compute stats i think im going to have 4 sections (projection, filter, sort, join, and aggregate, and other which will include everything else - getting more insight from others on potential execs to extract would be interesting). 

One of the open points is about how i aggregate stats on the same nodes / execs.  Im probably going to just rollup details per exec for v1 but you do lose some interesting information by doing that so ideally id like to improve on that.

An example of using this could be benchmarking and analyzing parquet files with different layouts (for example file sizes, row group sizes, bloom filters, page sizes, etc). Ideally running this command and comparing the results should make it easier to choose the optimal layout.

Todo for v1
- [x] Selectivity based on rows
- [x] Selectivity based on bytes
- [x] Add "Other" compute stats
- [x] Add "Join" compute stats
- [x] Add "Aggregate" compute stats
- [x] Cleanup display (alignment, same header widths, fixed decimals, etc)
- [x] ~For IO add % of total for scanning and opening~ Going to revisit this
- [x] ~Add throughput to summary based on output bytes from IO layer. Would be cool to do a pass that pulled actual file sizes that could be used.~ Going to revisit this
- [x] Ratio of selectivity to pruning for parquet
- [x] Avg/median time to scan single row group (total scanning time / row groups)
- [x] Update README

Future 
- identify nodes where some partitions are "hot" (i.e doing much more compute than others)

image

### CODE REVIEW

**Comment by matthewmturner** [comment]
@alamb FYI I've been working on adding this `--analyze` feature to the cli (its meant to complement the existing `EXPLAIN`, `EXPLAIN ANALYZE`, and `EXPLAIN VERBOSE` functionality - thought you may have some thoughts or ideas on it.

**Comment by alamb** [comment]
Thanks @matthewmturner  -- I'll try and check this out shortly

**Comment by matthewmturner** [comment]
Made some improvements - latest is in the below image.  Something I thought of that I think would be quite cool is adding a ratio of the queries selectivity to the effectiveness of its pruning to help guide improvements in how people layout their parquet files (for example end result is only a few rows but we do no row group pruning - that is far from ideal).

image

File: src/execution/stats.rs [hunk 1]
```diff
@@ -15,63 +15,729 @@
...
+    ///
+    /// V1: Simply look at row groups pruned by statistics and the row selectivity
+    /// TODO: Incorporate bloom filter and page index pruning
+    ///
+    /// Example calculations:
+    ///
+    /// 1. No pruning and select all rows
+    ///    - row groups pruned: 0
+    ///    - row groups matched: 10
+    ///    - row selectivity: 1.0
```
Comment (matthewmturner) [inline src/execution/stats.rs:106]: I am mocking out some calculations here to figure out how i want to compute "efficiency" - i just have a general idea that i want to take into account row selectivity and parquet pruning but need to flesh out the details and some tangible use cases.

Comment (matthewmturner) [reply src/execution/stats.rs:106]: Ended up making a [spreadsheet](https://docs.google.com/spreadsheets/d/1wDqe-oGHg9SDL_7VoIinvYoYV1Svst_5G5YTS2IFQ8g/edit?gid=0#gid=0) for this instead.

**Comment by matthewmturner** [comment]
This is the latest report thats produced.  Adding the throughput and IO as % of total then putting a bow on this with some docs and merging.

image

**Comment by matthewmturner** [comment]
I have also updated the PR description with more background context and examples how this could be used.

**Comment by matthewmturner** [comment]
Cleaned up the display a lot just now after testing some more complicated queries.  I think this is the output ill be going with for v1

image
